semantic_router/encoders/voyageai.py

```diff
@@ -0,0 +1,73 @@
+import os
+from time import sleep
+from typing import Any, List, Optional
+
+from pydantic.v1 import PrivateAttr
+
+from semantic_router.encoders import BaseEncoder
+from semantic_router.utils.defaults import EncoderDefault
+from semantic_router.utils.logger import logger
+
+
+class VoyageAIEncoder(BaseEncoder):
+    _client: Any = PrivateAttr()
+    type: str = "voyageai"
+
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        voyage_api_key: Optional[str] = None,
+        score_threshold: float = 0.82,
+    ):
+        if name is None:
+            name = EncoderDefault.VOYAGE.value["embedding_model"]
+        super().__init__(name=name, score_threshold=score_threshold)
+        self._client = self._initialize_client(api_key=voyage_api_key)
+
+    def _initialize_client(self, api_key: Optional[str] = None):
+        try:
+            import voyageai
+        except ImportError:
+            raise ImportError(
+                "Please install VoyageAI to use VoyageAIEncoder. "
+                "You can install it with: "
+                "`pip install 'semantic-router[voyageai]'`"
+            )
+
+        api_key = api_key or os.getenv("VOYAGEAI_API_KEY")
+        if api_key is None:
+            raise ValueError("VoyageAI API key not provided")
+        try:
+            client = voyageai.Client(api_key=api_key)
+        except Exception as e:
+            raise ValueError(f"Unable to connect to VoyageAI {e.args}: {e}") from e
```

Comment:
Just to help with the test problems, the test is looking to assert the error message outputs as `"VOYAGE API client failed to initialize. Error: Initialization error"` but because _e.args_ returns a tuple the output from the failure is actually `"Unable to connect to VoyageAI ('Initialization error',): Initialization error"`. 

To pass the assertion you could update your test assertion to match the args output or you could update the actual error message to `ValueError(f"VOYAGE API client failed to initialize. Error: {e}")` which should pass